Getting Started updates for v7

jenniferfell opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?
Documentation updates to the Getting Started (aka quickstart) document to:

- Make it current for v7
- Bring it up to current authoring style guidelines and practices
- Prepare it to integrate with content changes in PR https://github.com/angular/angular/pull/25776

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[X ] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

jenniferfell committed:
docs: getting started updated for accuracy and style

jenniferfell commented:
Reviewers: Changes coming when PR  #25776 lands or related PR from me: 
- Remove reference to Anatomy of the Setup. Replace with link to new workspace and file structure reference (in progress)
- Remove references and links to CLI wiki. We plan to do this across the doc set after filenames and titles for the new/updated stories and guides are more finalized.

jbogarthyde commented:
From Travis: Dangling Links Found in "guide/quickstart.json":
 - setup-systemjs-anatomy,
 - architecture

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -152,490 +150,39 @@ Open `src/app/app.component.css` and give the component some style.
-Whenever you want to know more about how Angular CLI works make sure to visit
-[the Angular CLI repository](https://github.com/angular/angular-cli) and
-[Wiki](https://github.com/angular/angular-cli/wiki).
+You can <a href="generated/zips/cli-quickstart/cli-quickstart.zip" target="_blank">download the example.</a> Most Angular guides contain links to associated example apps, so that you can see Angular concepts and code in action. 
```
anchor tag not closed

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -152,490 +150,39 @@ Open `src/app/app.component.css` and give the component some style.
-Whenever you want to know more about how Angular CLI works make sure to visit
-[the Angular CLI repository](https://github.com/angular/angular-cli) and
-[Wiki](https://github.com/angular/angular-cli/wiki).
+You can <a href="generated/zips/cli-quickstart/cli-quickstart.zip" target="_blank">download the example.</a> Most Angular guides contain links to associated example apps, so that you can see Angular concepts and code in action. 
```
Something wrong here - why is the rest of the text italic from here?

jenniferfell replied:
I don't see italic locally. Recent Mary Poppins (~30 min ago) doesn't show italic. Let me know if it creeps back in. Thanks!

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -152,490 +150,39 @@ Open `src/app/app.component.css` and give the component some style.
-  </tr>
-  <tr>
-    <td>
+For more details about Angular project files and the file structure, see [Anatomy of the setup](setup-systemjs-anatomy).
```
(guide/setup-systemjs-anatomy)

jenniferfell replied:
Done. Thx.

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -152,490 +150,39 @@ Open `src/app/app.component.css` and give the component some style.
 
-    </td>
-    <td>
+* The [Architecture guide](architecture "Architecture guide")  describes key concepts such as modules, components, services, dependency injection (DI). It provides a foundation for more in-depth guides about specific Angular concepts and features.  
```
(guide/architecture)

jenniferfell replied:
Done. Thx.

jenniferfell commented:
@jbogarthyde Thanks for showing me how Travis can help with link-checking. And for reading this!

Re the filename "systemjs", I wouldn't assume that the content applies to systemjs just because that's in the filename. At a glance, much of the structure seems similar to the one previously described in this getting started. So I think it was updated sometime in the Angular 2+ era. In any case, I think the right answer is still to replace that with your new workspace and project file reference (so we fix the filename issue and update the content).

mary-poppins commented:
You can preview 550bffc at https://pr26093-550bffc.ngbuilds.io/.

mary-poppins commented:
You can preview e98739c at https://pr26093-e98739c.ngbuilds.io/.

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -1,46 +1,46 @@
+You can download and install these npm packages with the [**npm client**](https://docs.npmjs.com/cli/install), which is installed with Node.js by default.
 
+The [**yarn client**](https://yarnpkg.com/en/) is a popular alternative for downloading and installing npm packages.
+The Angular CLI uses `yarn` by default to install npm packages when you create a new project.
```
the cli doesn't use yarn by default. it uses npm by default, but you can reconfigure it to use `ng config -g cli.packageManager yarn` to configure it to be the default, but I don't think we want to cover that in the quickstart.

jenniferfell replied:
Changed default to npm here. Opened Jira issue to make sure we describe how to change the npm package used by the CLI, probably in https://angular.io/guide/npm-packages

brandonroberts commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
-      They shouldn't be inside `src/` because e2e tests are really a separate app that
-      just so happens to test your main app.
-      That's also why they have their own `tsconfig.e2e.json`.
+These files go in the `root` folder next to `src/`.
```
Root of the project/workspace?

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -1,46 +1,46 @@
+
+The [**Angular CLI**](https://cli.angular.io/) is a **_command line interface_** tool
+that makes it easy to create a project, add files, and perform a variety of ongoing development tasks such as testing, bundling, and deployment.
+
```
You are using "project" throughout to refer to what is now a workspace, which contains what I guess is a "main" app (named "app", in the `app` subfolder).  You can add additional projects (apps or libraries), which get put into a `projects` subfolder. 

I'm finding this terminology confusing already. There is no *app* named "my-app". The root folder created with `ng new my-app` is named `my-app.  We are referring to it as "project-root"  -- that is, intuitively, what I would call it, but it is, in the current terminology, the *workspace* root - neither an app nor a project.

jenniferfell replied:
See notes above. Doc updated:
* workspace folder is called my-app
* initial app project is named my-app and resides in the src folder

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -51,9 +51,9 @@ Then install the [Angular CLI](https://github.com/angular/angular-cli) globally.
+{@a create-proj}
+
+## Step 2: Create a new project
 
```
This creates a new workspace, not a new project.

jenniferfell replied:
Creates: 
* a workspace
* two projects (an app project and an e2e project)

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -118,15 +117,14 @@ Your app greets you with a message:
 
```
root workspace directory?

jenniferfell replied:
I don't use "root" much. "Root workspace folder" and "workspace folder" seem the same to me, and one is shorter.

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
 
-An Angular CLI project is the foundation for both quick experiments and enterprise solutions.
+## Solution and project file review
 
```
Solution? solution to what?

jenniferfell replied:
Changed to match the tutorial: "Final code review".

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
-
-### The `src` folder
-Your app lives in the `src` folder.
+Your app resides in the `src` folder.
```
main app? root app? initial app? initial application files?
The generic name "app" for the folder (and as the default prefix for that app) just doesn't quite work with new multi-app workspace structure.

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
 
-    </td>
-    <td>
+* The [Architecture guide](guide/architecture "Architecture guide")  describes key concepts such as modules, components, services, dependency injection (DI). It provides a foundation for more in-depth guides about specific Angular concepts and features.  
```
services, *and* dependency...

jenniferfell replied:
Fixed.

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -87,10 +87,9 @@ See the [Angular Material documentation](https://material.angular.io/guides "Ang
 </div>
```
I don't think it's appropriate to mention `ng add` here. the readers are just getting started, and before they add more bells and whistles, we want them to get to the point where their app starts and they can see it. I suggest removing this or moving it to the very end if it fits there.

jenniferfell replied:
I'll remove. I was asked to put it here for v6 when ng add was new, but I agree that it's a distraction. Now that we have info about adding libraries in this doc set, I think we can just remove from here.

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
-### The root folder
-
-The `src/` folder is just one of the items inside the project's root folder.
 Other files help you build, test, maintain, document, and deploy the app.
```
can you combine this sentence with the previous one? both sentences describe the same thing but at different level of detail.

jenniferfell replied:
Simplified.

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
-      They shouldn't be inside `src/` because e2e tests are really a separate app that
-      just so happens to test your main app.
-      That's also why they have their own `tsconfig.e2e.json`.
+These files go in the `root` folder next to `src/`.
```
this also refers to the same info. please combine with previous two sentences.

jenniferfell replied:
Simplified

jbogarthyde commented on aio/content/guide/quickstart.md:
```diff
@@ -118,15 +117,14 @@ Your app greets you with a message:
 
```
Does it actually reflect the app name, or the workspace name? Does the command create an app named "my-app"? It doesn't look like it does.

jenniferfell replied:
It reflects the string that is in the app.component.ts file as the title="" property of AppComponent.

jenniferfell replied:
Everything below that is "welcome-app" is what I entered to ng new. Everything else is fixed.

directory structure:
welcome-app > src > app > app.component.ts

and in that file
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'welcome-app';
}

jenniferfell replied:
Image fixed.
Terminology TBD when we have signoff of definitions.

jenniferfell replied:
Image fixed. 
my-app is both the workspace folder name and the name of the initial app project.

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
+Now that you've seen the essentials of an Angular app and the Angular CLI, continue with these introductory materials: 
 
-      `.editorconfig`
+* The [Tour of Heroes tutorial](tutorial "Tour of Heroes tutorial") provides additional hands-on learning. It walks you through the steps to build an app that helps a staffing agency manage its stable of heroes. 
```
"stable of heroes" might be hard to understand for non-native speakers without additional context. maybe just say "manage its heroes."?

jenniferfell replied:
Simplified

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
 
-    </td>
-    <td>
+        - Acquiring and displaying a list of items (heroes)
```
drop "(heroes)" - it's not necessary here.

jenniferfell replied:
done

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
 
-    </td>
-    <td>
+        - Acquiring and displaying a list of items (heroes)
```
I think this list is indented too much which makes the list appear unformatted. try reindenting by two spaces from the `*` above and/or replace the leading `-` with `*`.

jenniferfell replied:
Not sure I'm still seeing what you saw. I see this. Ok?

![image](https://user-images.githubusercontent.com/31675481/46120525-5434c100-c1cd-11e8-8b47-7b4dc9329460.png)

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -147,495 +145,44 @@ Open `src/app/app.component.css` and give the component some style.
+* The [Architecture guide](guide/architecture "Architecture guide")  describes key concepts such as modules, components, services, dependency injection (DI). It provides a foundation for more in-depth guides about specific Angular concepts and features.  
 
-      Git configuration to make sure autogenerated files are not committed to source control.
+Both the Tutorial and Architecture guide are intended to give you the foundation to continue exploring Angular on your own, focusing on the features most important for your apps. 
```
drop the extraneous leading space

jenniferfell replied:
Seems fixed

IgorMinar reviewed (commented):
I left a few comments and suggestions.

I agree that the project files listing should not be in the quick guide, but I don't think that we should completely remove it since it contains some valuable information.

Could we move it elsewhere? Maybe standalone guide describing the project layout?

jenniferfell commented on aio/content/guide/quickstart.md:
```diff
@@ -1,46 +1,46 @@
 
-Good tools make application development quicker and easier to maintain than
-if you did everything by hand.
+Welcome to Angular! Angular makes it easy to build modern applications for the web, mobile, or desktop.  
```
Done.

jenniferfell committed:
fixup! docs: getting started updated for accuracy and style

jenniferfell commented:
Incorporated all feedback except:
- Terminology around workspace, project, and app. I'm going to wait until we have an approved set of definitions. 
- "app" v. "my-app" and the related picture of the running app. I'm tempted to rename the app to something more compelling than than my-app, such as angular-welcome. I'll have to see how much this unravels because we also have a downloadable solution.

jenniferfell committed:
fixup! docs: getting started updated for accuracy and style

jenniferfell committed:
fixup! docs: getting started updated for accuracy and style

jenniferfell commented on aio/content/guide/quickstart.md:
```diff
@@ -69,7 +69,7 @@ Generate a new project and default app by running the following command:
 
 
-The Angular CLI installs the necessary npm packages, creates the project files, and populates the project with a simple default app. This can take some time.
+The Angular CLI installs the necessary `npm` packages, creates the project files, and populates the project with a simple default app. This can take a few minutes.
```
New prompts? I don't get any new prompts. I'll need to redo this after code freeze, I guess.

jenniferfell replied:
New prompts mentioned. Thanks!

mary-poppins commented:
You can preview 95c15e5 at https://pr26093-95c15e5.ngbuilds.io/.

jenniferfell commented on aio/content/guide/quickstart.md:
```diff
@@ -118,15 +117,14 @@ Your app greets you with a message:
+This is the _root component_, and it is named `app-root`.
 You can find it in `./src/app/app.component.ts`.
 
 
```
I'm seeing ./src/app/app.component.ts no matter what string I feed into ng new.

mary-poppins commented:
You can preview 9f68bd5 at https://pr26093-9f68bd5.ngbuilds.io/.

jenniferfell committed:
fixup! docs: getting started updated for accuracy and style

mary-poppins commented:
You can preview 858aea2 at https://pr26093-858aea2.ngbuilds.io/.

jenniferfell commented on aio/content/guide/quickstart.md:
```diff
@@ -51,9 +51,9 @@ Then install the [Angular CLI](https://github.com/angular/angular-cli) globally.
+## Step 2: Create a new project
 
 
 
```
Done

jenniferfell committed:
fixup! docs: getting started updated for accuracy and style

jenniferfell committed:
fixup! docs: getting started updated for accuracy and style

jenniferfell commented:
I believe that I have addressed everyone's comments. 

Changes from v6-stable: 
* Simplified intro and added time estimate
* Added "Prerequisites" heading and merged in content from other locations, so this is the primary source of prereqs. Repositioned npm as the package manager used for the GS, not the only one supported. Updated node to require v8 or v10. 
* Step 1: Moved definition of the CLI to step 1, where CLI is installed. (The original version of this doc was a cli-specific quick start, so it made sense to open with a CLI definition.)
* Step 2: Rewrote almost completely to introduce basic definitions of workspace and project. Updated to be accurate for v7.
* Step 3: Fixed the image.
* Step 4: Added definition of component, expanded code section to show the entire component definition, expanded image of final Welcome screen. 
* Final code review: Replace old info about file structure with a code review similar to what the TOH Tutorial does. Moved download link here, so readers aren't as likely to "cheat" and download the app before doing the GS steps themselves. Removed file structure info, and added link to new workspace and project file doc.
* Next steps: Added architecture guide.

jenniferfell commented:
@IgorMinar @brandonroberts @jbogarthyde Any show-stoppers? Or just easy "while you're here" or "you mistyped" glitches? Thanks!

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
-  </tr>
-  <tr>
-    <td>
+These are the files that you modified in this Getting Started:
```
do we need this section? it seems to be not-essential.

jenniferfell replied:
Removed.

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
-      End-to-end test configuration for [Protractor](http://www.protractortest.org/),
-      used when running `ng e2e`.
+You can <a href="generated/zips/cli-quickstart/cli-quickstart.zip" target="_blank">download an example</a> of this modified Getting Started app. 
+Most Angular guides include links to download example files or run live examples in [Stackblitz](http://www.stackblitz.com), so that you can see Angular concepts and code in action. 
```
I don't think that this second sentence is relevant to the getting started guide. or is it? shouldn't we move this to the next steps section?

jenniferfell replied:
I included it here as a teaching moment about final code. One of the things that users learn in a getting started is how to learn...so it seemed to me like the right place to say "here is the final code for this doc, and you should look for examples with other docs."

Does it work better as a tip?

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
-</table>
 
-<div class="alert is-helpful">
+* The [Architecture guide](guide/architecture "Architecture guide")  describes key concepts such as modules, components, services, and dependency injection (DI). It provides a foundation for more in-depth guides about specific Angular concepts and features.  
```
double space before describes

jenniferfell replied:
good eye!

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
+* The [Architecture guide](guide/architecture "Architecture guide")  describes key concepts such as modules, components, services, and dependency injection (DI). It provides a foundation for more in-depth guides about specific Angular concepts and features.  
 
-### Next Step
+After the Tutorial and Architecture guide, you'll be ready to continue exploring Angular on your own, focusing on the features most important for your apps. 
```
"on your own" => "on your own via guides included in the Fundamentals, Techniques and other docs you can find in the site navigation"... or something like that...

jenniferfell replied:
good point. don't want them to feel we're sending them away
"you'll be ready to continue exploring Angular on your own through the other guides and references in this documentation set,..."

IgorMinar commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
+## Step 1: Install the Angular CLI
 
-Open a terminal window.
+The [**Angular CLI**](guide/glossary#command-line-interface-cli) is a **_command line interface_** tool
```
you've already defined this in the intro section above. can we dedupe?

jenniferfell replied:
Refocused a bit. Intro just talks about "the Angular CLI tool to accelerate development."

Here I refocused to teach a bit more about what it really can do for you: 
"You use the Angular CLI 
to create projects, generate application and library code, and perform a variety of ongoing development tasks such as testing, bundling, and deployment."

It seemed a bit much to put this earlier, but I think it's a differentiator from other platforms, so I didn't want to lose it. 

Better?

IgorMinar reviewed (approved):
This looks pretty good. I've just asked for some minor changes.

jenniferfell commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
-Install **[Node.js® and npm](https://nodejs.org/en/download/)**
-if they are not already on your machine.
+This guide takes less than 30 minutes to complete. 
+At the end of this guide, you can download a copy of the final application code.
```
Added note about downloading even if you don't do this QS. 

I didnt offer a download for each step because the steps are so small. Usually we only offer interim downloads if the steps are huge, such as in the TOH, and someone might reasonably want to start/resume in the middle with a clean download. Here there's actually nothing different from what ng new creates until after you change the component. And then you're done.

jenniferfell replied:
@IgorMinar P.S. I'm not sure that we want completely new users to even consider downloading the example before they do this guide. The link used to be here at the top, and I moved it to the bottom to force them to at least do the setup and change one component before downloading code. More advanced users who come back will know they can skip to the end and download.

jenniferfell commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
-application in TypeScript, using the Angular CLI
-while adhering to the [Style Guide](guide/styleguide) recommendations that
+This guide shows you how to build and run a simple Angular
+app. You'll use the [Angular CLI](cli "CLI command reference") to accelerate development, 
```
done

jenniferfell commented on aio/content/guide/quickstart.md:
```diff
@@ -1,641 +1,185 @@
+
+## Step 2: Create a workspace and initial application
+
+You develop apps in the context of an Angular [**workspace**](guide/glossary#workspace). A workspace contains the files for one or more [**projects**](guide/glossary/#project). A project is the set of files that comprise an app, a library, or end-to-end (e2e) tests. 
```
Popped her a note. :-)

jenniferfell committed:
fixup! docs: getting started updated for accuracy and style
